ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -126,12 +128,19 @@ void handleConnect(final EthPeer peer) {
     for (final Transaction transaction : localTransactions) {
       peerTransactionTracker.addToPeerSendQueue(peer, transaction);
     }
-    peerPendingTransactionTracker.ifPresent(
+    maybePeerPendingTransactionTracker.ifPresent(
         peerPendingTransactionTracker -> {
           if (peerPendingTransactionTracker.isPeerSupported(peer, EthProtocol.ETH65)) {
-            final Collection<Hash> hashes = getNewPooledHashes();
-            for (final Hash hash : hashes) {
-              peerPendingTransactionTracker.addToPeerSendQueue(peer, hash);
+            Iterator<Hash> hashIterator = getNewPooledHashes().iterator();
```

Comment:
I'm assuming there is some reason to go from a for loop iteration to a direct iterator.  Is it because the iterator handles the unstable collection better?

Reply:
As far as I know you can't catch the exception and continue over the syntactic sugar iterator.